yellowbrick/classifier/class_balance_heat_map.py

```diff
@@ -0,0 +1,169 @@
+# Author:   Benjamin Bengfort <bbengfort@districtdatalabs.com>
+# Author:   Neal Humphrey
+# Author:   Larry Gray
+# Created:  Sun March 4 20:45:00 2018 -0400
+#
+# Copyright (C) 2017 District Data Labs
+# For license information, see LICENSE.txt
+#
+# ID: class_balance_heat_map.py [5388065] lwgray@gmail.com $
+
+"""
+Class balance heat map visualizer for showing per-class support.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import numpy as np
+
+from .base import ClassificationScoreVisualizer
+
+from sklearn.model_selection import train_test_split
+from sklearn.utils.multiclass import unique_labels
+
+
+##########################################################################
+## Class Balance Heat Map Chart
+##########################################################################
+
+class ClassBalanceHeatMap(ClassificationScoreVisualizer):
+    """
+    Class balance heat map that shows the support for each class in the
+    fitted classification model displayed as a stacked bar.  Each bar is
+    segmented to show the distribution of predicted classes for each
+    class. It is initialized with a fitted model and generates a
+    class balance heat map chart on draw.
+
+    Parameters
+    ----------
+    ax: axes
+        the axis to plot the figure on.
+    model: estimator
+        Scikit-Learn estimator object. Should be an instance of a classifier,
+        else ``__init__()`` will raise an exception.
+    classes: list
+        A list of class names for the legend. If classes is None and a y value
+        is passed to fit then the classes are selected from the target vector.
+    kwargs: dict
+        Keyword arguments passed to the super class. Here, used
+        to colorize the bars in the histogram.
+    Notes
+    -----
+    These parameters can be influenced later on in the visualization
+    process, but can and should be set as early as possible.
+    """
+
+    def score(self, X, y=None, **kwargs):
+        """
+        Generates a 2D array where each row is the count of the
+        predicted classes and each column is the true class
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        Returns
+        -------
+
+        ax : the axis with the plotted figure
+        """
+
+        y_pred = self.predict(X)
+        self.indices = unique_labels(y, y_pred)
+        self.classes_ = self.classes_ if self.classes_ is not None else self.indices
+        self.scores = np.array([
+            [(y_pred[y == label_t] == label_p).sum()
+            for label_p in self.indices]
+            for label_t in self.indices
```

Comment:
minor request to break this up nested list comprehension up as it is a bit hard to read. at a minimum, a nice comment explaining it would be great

Reply:
I am working on it... it was left over from @bbengfort  and I am still having trouble decoding it.

Reply:
This is setting up a two-dimensional array, so I think it's going to be tough to break it up. Might I suggest the following?

```python
# Create a table of scores whose rows are the true classes
# and whose columns are the predicted classes; each element
# is the count of predictions for that class that match the true 
# value of that class. 
self.scores_ = np.array([
    [
        (y_pred[y == label_t] == label_p).sum()
        for label_p in self.indices
    ]
    for label_t in self.indices
])
```

Reply:
@bbengfort :+1:

Reply:
@ndanielsen should i simply create my own folder in the examples directory?

Reply:
You can, using your GitHub username preferably, but that's not required. Documentation for this would go in `docs/api/classification/class_balance.rst` or you could make an .rst file next to that if you prefered.